monai/transforms/spatial/array.py

```diff
@@ -3128,8 +3128,16 @@ def __call__(self, array: NdarrayOrTensor):
             patched_image, locations = self.filter_threshold(patched_image, locations)
 
         # Convert to original data type
+        # Locations data type should differ from array data type
+        if isinstance(array, torch.Tensor):
+            location_dtype = torch.int
+        elif isinstance(array, np.ndarray):
+            location_dtype = np.int
+        else:
+            location_dtype = None
         output = list(
-            zip(convert_to_dst_type(src=patched_image, dst=array)[0], convert_to_dst_type(src=locations, dst=array)[0])
+            zip(convert_to_dst_type(src=patched_image, dst=array)[0],
+                convert_to_dst_type(src=locations, dst=array, dtype=location_dtype)[0])
```

Comment:
thanks for the patch, I think 
```py
convert_to_dst_type(src=locations, dst=array, dtype=int)[0]
```
is good enough in this case, because the function will handle `dtype` properly based on the `dst`:

```py
>>> convert_to_dst_type(torch.zeros((2,2)), dst=numpy.zeros((2, 2)), dtype=int)[0]
array([[0, 0],
       [0, 0]])
>>> convert_to_dst_type(torch.zeros((2,2)), dst=torch.zeros((2,2)), dtype=int)[0]
tensor([[0, 0],
        [0, 0]])
>>> convert_to_dst_type(numpy.zeros((2,2)), dst=torch.zeros((2,2)), dtype=int)[0]
tensor([[0, 0],
        [0, 0]])
>>> convert_to_dst_type(numpy.zeros((2,2)), dst=numpy.zeros((2,2)), dtype=int)[0]
array([[0, 0],
       [0, 0]])
```

Reply:
Brilliant - I updated accordingly